### ISSUE TITLE
About your local module

### ISSUE THREAD

**Issue by CR7forMadrid** [Fried-Rice-Lab/EMT#3]
One more question: Why is your proposed Local Transformer Layer implemented through BN+PM? What is the rationale for doing so?  Wouldn't the BN layer have a negative impact on the SR task?  What if PM alone is used?

**Comment by Jinchen2028** [comment]
Thank you very much for your interest in our work: 
Based on what I know, I believe the BN layer you mentioned can help the network train better. In addition, how to understand the code, please read the abstract.

**Comment by CR7forMadrid** [comment]
```
class TokenMixer(nn.Module):
    def __init__(self, dim: int) -> None:
        super().__init__()

        self.token_mixer = PixelMixer(planes=dim)
        self.norm = nn.BatchNorm2d(dim)

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        return self.norm(self.token_mixer(x) - x)
```
The Local Block implemented in the code you provided does not match the diagram provided in your paper.  Perhaps you can tell me which paper you got the idea to build Local Block from? Because BN proposed in EDSR that BN layer normalizes all features in a batch of samples and will ignore the absolute differences between features and only consider the relative differences. Moreover, using BN layer would normalize the image color distribution and destroy the contrast information of the image, while BN layer needs to consume the same size of memory as its connected convolution layer.

**Comment by liozur** [comment]
Thank you for your attention.

1. For the use of BN, please refer to our previous work "[Image Super-Resolution using Efficient Striped Window Transformer](https://github.com/Fried-Rice-Lab/FriedRiceLab#our-works)". Experimentally, we proved that for SR models with limited capacity (most LSR models), using BN can better improve the generalization of the model.

2. Please check our [PM](https://github.com/Fried-Rice-Lab/FriedRiceLab/blob/master/archs/utils/_tool.py) module carefully. In the implementation, we manipulate the kernel of the convolution layer to implement the shift operation.

Please feel free to reopen this issue or create a new one if you have any questions.

:)
